js/poloniex.js

```diff
@@ -683,15 +683,25 @@ module.exports = class poloniex extends Exchange {
             response = await this.privatePostCancelOrder (this.extend ({
                 'orderNumber': id,
             }, params));
-            if (id in this.orders)
-                this.orders[id]['status'] = 'canceled';
         } catch (e) {
-            if (this.last_http_response) {
-                if (this.last_http_response.indexOf ('Invalid order') >= 0)
-                    throw new OrderNotFound (this.id + ' cancelOrder() error: ' + this.last_http_response);
+            if (e instanceof CancelPending) {
```

Comment:
This instanceof will cause some troubles with the the transpiler... The flow of RTTI for exceptions in JS is different from Python/PHP/C++, where type matching is done within the catch clause. This is why we can't use instanceof like that, unfortunately... Unless we add a case for it to the transpiler. Let me investigate, plz. Will get back to you on this shortly.

Reply:
Oops... Yeah, you told me that already, I completely forgot (

Reply:
No worries, not having instanceof at hand is counter-intuitive indeed. I know that we can transpile this to Python, not sure about PHP yet, looking up.

Reply:
If it won't we'll be able to go with something like `this.instanceOf (exception, class)`, what do you think?

Reply:
@mkutny sure! I'm still determined to transpile it to natural language syntax, but we have alternative options just in case.